In [1]:
import pandas as pd
import numpy as np
from bs4 import BeautifulSoup #Importing BS4 Beta
import requests

In [73]:
site = 'https://mcphd-tx.org/Covid-19-Case-Updates/'
page = requests.get(site)
soup = BeautifulSoup(page.content, 'html.parser')

In [74]:
print(soup.prettify())

<!DOCTYPE html>
<!--[if IE 7]>
<html class="ie ie7" lang="en-US">
<![endif]-->
<!--[if IE 8]>
<html class="ie ie8" lang="en-US">
<![endif]-->
<!--[if !(IE 7) | !(IE 8)  ]><!-->
<html lang="en-US">
 <!--<![endif]-->
 <!--[if IE 9]>
<style>.main_nav ul li{ margin:0 6px !important;}
.main_nav ul li a { font-family:Arial !important; }
ul.nav-menu li:hover > ul, .nav-menu ul li:hover > ul{ padding-top:20px !important;}
.nav-menu .sub-menu{ left:-5px !important;}
</style>
<![endif]-->
 <head>
  <meta content="IE=edge" http-equiv="X-UA-Compatible"/>
  <meta charset="utf-8"/>
  <meta content="width=device-width" name="viewport"/>
  <title>
   Covid-19 Case Updates | Public Health District
  </title>
  <link href="http://gmpg.org/xfn/11" rel="profile"/>
  <link href="https://mcphd-tx.org/xmlrpc.php" rel="pingback"/>
  <link href="" rel="shortcut icon"/>
  <!--[if lt IE 9]>
	<script src="https://mcphd-tx.org/wp-content/themes/mchd/js/html5.js"></script>
	<![endif]-->
  <link href="https://mcphd-

In [75]:
test = soup.findAll('tbody')

In [76]:
test[0].contents[1].contents[1].contents[0]

'77356'

Very ugly nesting structure if I use this hierarchy.

In [77]:
test[0].contents[-2]

<tr class="row-1429 odd">
<td class="column-1">77354</td><td class="column-2">1428</td><td class="column-3">Male</td><td class="column-4">50-59</td><td class="column-5">Magnolia</td><td class="column-6">Active</td>
</tr>

In [78]:
len(test[0].contents[-2])

8

In [79]:
for i, n in enumerate(test[0].contents[-2]):
    print(i, n)

0 

1 <td class="column-1">77354</td>
2 <td class="column-2">1428</td>
3 <td class="column-3">Male</td>
4 <td class="column-4">50-59</td>
5 <td class="column-5">Magnolia</td>
6 <td class="column-6">Active</td>
7 



It looks like even though displaying all the data requires a button press, the HTML may have all the data already? Since our last case displayed is number 1428, and the limit is 1000 cases at a time.

In [80]:
for i, n in enumerate(test[0].contents[1]):
    print(i, n)

0 

1 <td class="column-1">77356</td>
2 <td class="column-2">1</td>
3 <td class="column-3">Male</td>
4 <td class="column-4">40-49</td>
5 <td class="column-5">Montgomery</td>
6 <td class="column-6">Recovered</td>
7 



So, it looks like instead of selectively loading in data, they're loading in all the data and then using HTML to selectively obscure it. This means I won't need an extra tool to extra the entire dataset.

In [81]:
test2 = soup.findAll("td")

In [82]:
len(test2)

8568

In [83]:
for i in range(7):
    print(test2[i].contents)
    print(test2[i].contents[0])

['77356']
77356
['1']
1
['Male']
Male
['40-49']
40-49
['Montgomery']
Montgomery
['Recovered']
Recovered
['77386']
77386


In [58]:
test2[0].contents[0]

'77356'

In [90]:
soup.td.contents

['77356']

In [68]:
copy.unwrap()

ValueError: Cannot replace an element with its contents when thatelement is not part of a tree.